core/trashcan.js

```diff
@@ -506,7 +506,8 @@ Blockly.Trashcan.prototype.position = function(metrics, savedPositions) {
 /**
  * Returns the bounding rectangle of the UI element in pixel units relative to
  * the Blockly injection div.
- * @return {!Blockly.utils.Rect} The UI elements’s bounding box.
+ * @return {?Blockly.utils.Rect} The UI elements’s bounding box. Null if
+ *   bounding box should be ignored by other UI elements.
```

Comment:
Note: I _think_ you could have left the type signature alone here, since `{!Type}` is a subtype of `{?Type}`.  It would avoid breaking someone's code in the (unlikely!) event that they were depending on the return value of `Trashcan.p.getBoundingRectangle` being non-nullable but don't care about the interface.

(Same for `ZoomControls`, obviously.)

Reply:
I didn't realize it was a subtype, that's interesting.
This method was introduced as part of the `IPositionable` api (which hasn't yet been widely advertised), so its *probably* unlikely that someone is depending on.

I'd like to make the argument that making this nullable gives some extra flexibility if we, or a plugin, wanted to make the trashcan hide-able (where it might make sense for them to have this return null).